pkg/operator/kube_cloud_config/aws.go

```diff
@@ -0,0 +1,100 @@
+package kube_cloud_config
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"gopkg.in/gcfg.v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/legacy-cloud-providers/aws"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+func awsTransformer(input *corev1.ConfigMap, key string, infra *configv1.Infrastructure) (*corev1.ConfigMap, error) {
+	if !(infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.AWSPlatformType) {
+		return nil, fmt.Errorf("invalid platform, expected to be AWS")
+	}
+	if !(infra.Status.PlatformStatus.AWS != nil && len(infra.Status.PlatformStatus.AWS.ServiceEndpoints) > 0) {
+		return asIsTransformer(input, key, infra) // no transformation required
+	}
+
+	var region string
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil {
+		region = infra.Status.PlatformStatus.AWS.Region
+	}
+	if region == "" {
+		return nil, fmt.Errorf("Required region for AWS platform infrastructures.%s/cluster not found", configv1.GroupName)
+	}
+
+	output := input.DeepCopy()
```

Comment:
update namespace here to avoid overwrites.

Reply:
> https://github.com/openshift/cluster-config-operator/pull/120/commits/626e8aae7dc1ccee9268753fd17744507e2d2603#diff-99d1c645e7848874c8b0c0ec171e00e1R27-R29

the transformer states that only the data and binaryData will be used by the consumer. the consumer has control over those fields.

Reply:
> > [626e8aa#diff-99d1c645e7848874c8b0c0ec171e00e1R27-R29](https://github.com/openshift/cluster-config-operator/commit/626e8aae7dc1ccee9268753fd17744507e2d2603#diff-99d1c645e7848874c8b0c0ec171e00e1R27-R29)
> 
> the transformer states that only the data and binaryData will be used by the consumer. the consumer has control over those fields.

belt and suspenders please